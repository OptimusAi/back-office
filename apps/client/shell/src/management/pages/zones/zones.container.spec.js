import 'regenerator-runtime/runtime';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Zones from './zones.container';
import configureStore from 'redux-mock-store';
import SpecFriendly from '../../../spec-friendly';

const mockStore = configureStore([]);

describe('Zones Container', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      app: {
        zones: {
          ids: [1],
          entities: {
            1: {
              id: 1,
              zone: 1,
              latitude: 80.40821857,
              longitude: -110.87463,
              pairingTimeLimit: 15,
              arrivalGraceTime: 5,
              departureGraceTime: 10,
              zoneCategory: {
                id: 'z001',
                name: 'On Street',
                description: 'On street parking',
              },
              city: {
                id: '1',
                city: 'Calgary',
                countrySubdivision: {
                  name: 'ALBERTA',
                  code: 'AB',
                  countryCode: 'CA',
                  country: 'Canada',
                  regionType: 'PROVINCE',
                },
              },
            },
          },
        },
      },
      management: {
        zones: {
          activeZoneId: 1,
          zoneCategoriesList: [{ id: 1 }],
          zoneCitiesList: [{ id: 1 }],
        },
      },
      permissions: {
        permissions: [],
      },
    });

    component = renderer.create(
      <SpecFriendly>
        <Provider store={store}>
          <Zones />
        </Provider>
      </SpecFriendly>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
