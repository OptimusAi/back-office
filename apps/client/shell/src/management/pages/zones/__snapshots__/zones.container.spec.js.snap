// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zones Container should render with given state from Redux store 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="makeStyles-gridToolbar-2"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Zones
    </h5>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeSmall"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        Add Zone
      </span>
    </button>
  </div>
  <div
    className="makeStyles-gridContainer-3"
  >
    <div
      aria-colcount={9}
      aria-multiselectable={false}
      aria-rowcount={2}
      className="MuiDataGrid-root MuiDataGrid-root--densityStandard css-r11z79-MuiDataGrid-root"
      role="grid"
    >
      <div
        className="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          className="MuiDataGrid-overlay css-18rq0oa-MuiDataGrid-overlay"
          error={[TypeError: Cannot set property 'scrollLeft' of null]}
        >
          An error occurred.
        </div>
      </div>
    </div>
  </div>
</div>
`;
